dataprofiler/profilers/profile_builder.py

```diff
@@ -1550,6 +1565,18 @@ def tqdm(l):
         # Newly introduced features (python3.8) improves the situation
         sample_ids = np.array(sample_ids)
 
+        # Record the previous mean/std values of columns that need to
+        # have correlation updated. Non-numeric columns have NaN
+        # so that they stay NaN when updating.
+        prev_dependent_properties = {
+            'mean': np.full(len(self._profile.keys()), np.nan),
+            'std': np.full(len(self._profile.keys()), np.nan)
+        }
+        for id, col in enumerate(self._profile.keys()):
+            if self._profile[col].profile['data_type'] in ['int', 'float']:
```

Comment:
This line causes some test cases to fail with a KeyError on "statistics". Some test cases mock out the compilers, so the profile property can't be called without an error.